### PR TITLE
ERMS-2338: Überarbeitung Titel Umfrage + Bug Delete Survey

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy [hunk 1]
```diff
@@ -2056,39 +2057,35 @@ class SurveyController {
 
                     SurveyConfig.findAllBySurveyInfo(surveyInfo).each { config ->
 
-                        config.surveyInfo = null
                         DocContext.findAllBySurveyConfig(config).each {
-                            it.delete()
+                            it.delete(flush: true)
```
Comment (croqueGrec09) [inline app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:2061]: Needed?

File: app/grails-app/views/subscription/addEntitlements.gsp [hunk 2]
```diff
@@ -198,45 +198,57 @@ ${message(code: 'subscription.details.availableTitles')} ( ${message(code: 'defa
             <strong><g:link controller="title" action="show"
                             id="${tipp.title.id}">${tipp.title.title}</g:link></strong>
 
-            <g:if test="${tipp?.title instanceof BookInstance }">
-                (${message(code: 'title.volume.label')} ${tipp?.title?.volume})
+            <g:if test="${tipp?.title instanceof BookInstance && tipp?.title?.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/subscription/addEntitlements.gsp:201]: to be or not to be => ?

File: app/grails-app/views/subscription/addEntitlements.gsp [hunk 3]
```diff
@@ -198,45 +198,57 @@ ${message(code: 'subscription.details.availableTitles')} ( ${message(code: 'defa
...
+                (${message(code: 'title.volume.label')} ${tipp.title.volume})
             </g:if>
 
             <g:if test="${tipp?.title instanceof BookInstance && (tipp?.title?.firstAuthor || tipp?.title?.firstEditor)}">
                 <br><b>${tipp?.title?.getEbookFirstAutorOrFirstEditor()}</b>
```
Comment (croqueGrec09) [inline app/grails-app/views/subscription/addEntitlements.gsp:206]: to be or not to be => ?

File: app/grails-app/views/subscription/showEntitlementsRenewWithSurvey.gsp [hunk 4]
```diff
@@ -96,63 +96,59 @@
...
-                               data-content="${tipp?.platform.name}"
-                               href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                               target="_blank"><i class="cloud icon"></i></a>
+                            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
                         </g:if>
                         <br>
 
                         <div class="la-icon-list">
-                            <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance }">
+                            <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/subscription/showEntitlementsRenewWithSurvey.gsp:104]: to be or not to be => ?

File: app/grails-app/views/survey/showEntitlementsRenew.gsp [hunk 5]
```diff
@@ -139,27 +139,23 @@
...
-                               data-content="${tipp?.platform.name}"
-                               href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                               target="_blank"><i class="cloud icon"></i></a>
+                            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
                         </g:if>
                         <br>
 
                         <div class="la-icon-list">
-                            <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance}">
+                            <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/survey/showEntitlementsRenew.gsp:147]: to be or not to be => ?

File: app/grails-app/views/templates/survey/_entitlementTableSurvey.gsp [hunk 6]
```diff
@@ -52,78 +52,59 @@
...
-                           data-content="${tipp?.platform.name}"
-                           href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                           target="_blank"><i class="cloud icon"></i></a>
+                        <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
                     </g:if>
                     <br>
 
                     <div class="la-icon-list">
-                        <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance}">
+                        <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/survey/_entitlementTableSurvey.gsp:60]: to be or not to be => ?

File: app/grails-app/views/templates/tipps/_entitlementTable.gsp [hunk 7]
```diff
@@ -59,65 +59,50 @@
...
-                                <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-                                   data-content="${tipp?.platform.name}"
-                                   href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                                   target="_blank"><i class="cloud icon"></i></a>
+                                <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
                             </g:if>
                             <br>
                             <div class="la-icon-list">
-                                <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance }">
+                                <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/tipps/_entitlementTable.gsp:66]: to be or not to be => ?

File: app/grails-app/views/templates/tipps/_table.gsp [hunk 8]
```diff
@@ -31,54 +31,50 @@
...
-                    <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-                       data-content="${tipp?.platform.name}"
-                       href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                       target="_blank"><i class="cloud icon"></i></a>
+                    <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
                 </g:if>
                 <br>
                 <div class="la-icon-list">
-                    <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance }">
+                    <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.volume}">
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/tipps/_table.gsp:38]: to be or not to be => ?

File: app/grails-app/views/templates/tipps/_table.gsp [hunk 9]
```diff
@@ -31,54 +31,50 @@
...
                             <i class="grey icon user circle la-popup-tooltip la-delay" data-content="${message(code: 'author.slash.editor')}"></i>
                             <div class="content">
-                                ${tipp?.title?.getEbookFirstAutorOrFirstEditor()}
+                                ${tipp?.title.getEbookFirstAutorOrFirstEditor()}
                             </div>
                         </div>
                     </g:if>
 
-                    <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance}">
+                    <g:if test="${tipp?.title instanceof com.k_int.kbplus.BookInstance && tipp?.title.editionStatement}">
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/tipps/_table.gsp:56]: to be or not to be => ?

File: app/grails-app/views/public/gascoDetails.gsp [hunk 10]
```diff
@@ -73,10 +73,7 @@
                         <br />
 
                         <g:if test="${tipp.hostPlatformURL}">
-                            <a href="${tipp.hostPlatformURL}" title="${tipp.hostPlatformURL}" target="_blank">
-                                ${tipp.hostPlatformURL}
-                                <i class="ui icon share square"></i>
-                            </a>
+                            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/public/gascoDetails.gsp:76]: use tipp.hostPlatformURL.startsWith('http')

File: app/grails-app/views/tipp/show.gsp [hunk 11]
```diff
@@ -162,11 +162,7 @@
         </div>
         <br>
         <g:if test="${tipp.hostPlatformURL}">
-            <a role="button" class="ui icon tiny blue button la-js-dont-hide-button la-popup-tooltip la-delay"
-            <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-               data-content="${tipp?.platform.name}"
-               href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-               target="_blank"><i class="cloud icon"></i></a>
+            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/tipp/show.gsp:165]: see above

File: app/grails-app/views/templates/tipps/_table.gsp (hunk 8 above)
Comment (klodav) [inline app/grails-app/views/templates/tipps/_table.gsp:38]: multiple tipp?. following

File: app/grails-app/views/public/gascoDetailsIssueEntitlements.gsp [hunk 12]
```diff
@@ -73,10 +73,7 @@
                         <br />
 
                         <g:if test="${tipp.hostPlatformURL}">
-                            <a href="${tipp.hostPlatformURL}" title="${tipp.hostPlatformURL}" target="_blank">
-                                ${tipp.hostPlatformURL}
-                                <i class="ui icon share square"></i>
-                            </a>
+                            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/public/gascoDetailsIssueEntitlements.gsp:76]: see above

File: app/grails-app/views/subscription/index.gsp [hunk 13]
```diff
@@ -244,12 +244,7 @@
                                             action="show"><strong>${ie.tipp.title.title}</strong>
                                     </g:link>
                                     <g:if test="${ie.tipp.hostPlatformURL}">
-                                        <a role="button" class="ui icon tiny blue button la-js-dont-hide-button la-popup-tooltip la-delay"
-                                        <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-                                           data-content="${ie.tipp.platform.name}"
-
-                                           href="${ie.tipp.hostPlatformURL?.contains('http') ? ie.tipp.hostPlatformURL : 'http://' + ie.tipp.hostPlatformURL}"
-                                           target="_blank"><i class="cloud icon"></i></a>
+                                        <semui:linkIcon href="${ie.tipp.hostPlatformURL.contains('http') ? ie.tipp.hostPlatformURL : 'http://' + ie.tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/subscription/index.gsp:247]: see above

File: app/grails-app/views/subscription/showEntitlementsRenewWithSurvey.gsp [hunk 14]
```diff
@@ -96,63 +96,59 @@
                                         id="${tipp.title.id}">${tipp.title.title}</g:link></strong>
 
                         <g:if test="${tipp.hostPlatformURL}">
-                            <a role="button" class="ui icon tiny blue button la-js-dont-hide-button la-popup-tooltip la-delay"
-                            <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-                               data-content="${tipp?.platform.name}"
-                               href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"
-                               target="_blank"><i class="cloud icon"></i></a>
+                            <semui:linkIcon href="${tipp.hostPlatformURL.contains('http') ? tipp.hostPlatformURL : 'http://' + tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/subscription/showEntitlementsRenewWithSurvey.gsp:99]: see above

File: app/grails-app/views/survey/surveyTitles.gsp [hunk 15]
```diff
@@ -125,12 +125,7 @@
                                 action="show"><strong>${ie.tipp.title.title}</strong>
                         </g:link>
                         <g:if test="${ie.tipp.hostPlatformURL}">
-                            <a role="button" class="ui icon tiny blue button la-js-dont-hide-button la-popup-tooltip la-delay"
-                            <%-- data-content="${message(code: 'tipp.tooltip.callUrl')}" --%>
-                               data-content="${ie.tipp.platform.name}"
-
-                               href="${ie.tipp.hostPlatformURL.contains('http') ? ie.tipp.hostPlatformURL : 'http://' + ie.tipp.hostPlatformURL}"
-                               target="_blank"><i class="cloud icon"></i></a>
+                            <semui:linkIcon href="${ie.tipp.hostPlatformURL.contains('http') ? ie.tipp.hostPlatformURL : 'http://' + ie.tipp.hostPlatformURL}"/>
```
Comment (klodav) [inline app/grails-app/views/survey/surveyTitles.gsp:128]: see above

File: app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy (hunk 1 above)
Comment (Moetez16) [reply app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:2061]: Yes

**Pull request merged**